services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/Persistence.kt

```diff
@@ -255,6 +250,67 @@ interface HankeRepository : JpaRepository<HankeEntity, Int> {
     */
 }
 
+@Entity
+@Table(name = "haittojenhallinta")
+class HaittojenHallintaEntity(
+    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Long? = null,
+
+    @ElementCollection
+    @CollectionTable(
+            name = "haittojenhallinta_kuvaukset",
+            joinColumns = [JoinColumn(name = "haittojenhallinta_id", referencedColumnName = "id")])
+    @MapKeyJoinColumn(name = "key")
+    @MapKeyEnumerated(EnumType.STRING)
+    @Column(name = "value")
+    var kuvaukset: MutableMap<HaittojenHallintaKentta, String?> = mutableMapOf(),
+
+) {
+
+    companion object {
+        fun toEntity(source: HaittojenHallinta?, target: HaittojenHallintaEntity) {
+            if (source == null) {
+                return;
+            }
+            target.kuvaukset.clear()
+            source.kuvaukset.forEach {
+                if (it.value.kaytossaHankkeessa) {
+                    target.kuvaukset[it.key] = it.value.kuvaus
+                }
+            }
+        }
+
+        fun toDto(source: HaittojenHallintaEntity, target: HaittojenHallinta) {
```

Comment:
Mulla tulee tästä poikkeus:
```
java.lang.NullPointerException: Parameter specified as non-null is null: method fi.hel.haitaton.hanke.HaittojenHallintaEntity$Companion.toDto, parameter source
	at fi.hel.haitaton.hanke.HaittojenHallintaEntity$Companion.toDto(Persistence.kt)
```

Epäilen, että johtuu siitä, että mulla on jo valmiiksi kannassa hanke, mutta sillä ei tietenkään ole vielä HaittojenHallinta-viittausta.

Jos on tarkoitus, että kaikkiin hankkeisiin aina liittyy toi haittojenhallinta-taulun rivi, niin se viiteavain pitäisi olla NOT NULL ja migraatiossa pitäisi luoda sellainen kaikille hankkeille.

Reply:
Onko tälle tehty jotain?